### PR TITLE
Add build config prevent_unsafe_narrowing to support abseil roll

### PR DESCRIPTION
**Pull request by dnfield**
Related to https://flutter-review.googlesource.com/c/third_party/abseil-cpp/+/52880

Chromium added this config, we're missing it, newer versions of abseil want it.

### CODE REVIEW

File: build/config/compiler/BUILD.gn [hunk 1]
```diff
@@ -998,3 +998,43 @@ config("no_symbols") {
...
+#   configs += [ "//build/config/compiler:prevent_unsafe_narrowing" ]
+config("prevent_unsafe_narrowing") {
+  if (is_clang) {
+    cflags = [
+      "-Wshorten-64-to-32",
+      "-Wimplicit-int-conversion",
+      "-Wsign-compare",
+      "-Wsign-conversion",
+    ]
+    if (!is_nacl) {
```
Comment (chinmaygarde) [inline build/config/compiler/BUILD.gn:1032]: You can get rid of the `is_nacl`. That is never going to be relevant.

**Review by chinmaygarde — APPROVED** [review]

**Review by matanlurey — APPROVED** [review]

**Comment by matanlurey** [comment]
Thanks for the awesome detailed comment!

**Comment by dnfield** [comment]
> Thanks for the awesome detailed comment!

That is verbatim fro mthe Chromium build repo :D

File: build/config/compiler/BUILD.gn (hunk 1 above)
Comment (dnfield) [reply build/config/compiler/BUILD.gn]: Done

**Comment by matanlurey** [comment]
> > Thanks for the awesome detailed comment!
> 
> That is verbatim fro mthe Chromium build repo :D

Even better, great copy pasting!

**Pull request merged**